Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptEditController.java

```diff
@@ -180,11 +146,39 @@ public String confirmlEdit(@PathVariable("id") String id,
 	 * @return The list of existing concepts
 	 */
 	@RequestMapping(method = RequestMethod.GET, value = "conceptEditSynonymView")
-	public @ResponseBody
-	ConceptEntry[] searchConcept(@RequestParam("synonymname") String synonymname) {
-		ConceptEntry[] entries = conceptManager
-				.getConceptListEntriesForWord(synonymname.trim());
-		return entries;
+	public @ResponseBody ResponseEntity<String> searchConcept(@RequestParam("synonymname") String synonymname) {
+		ConceptEntry[] entries = conceptManager.getConceptListEntriesForWord(synonymname.trim());
+
+		StringBuffer jsonStringBuilder = new StringBuffer("{");
+		jsonStringBuilder.append("\"Total\":");
+		jsonStringBuilder.append(entries.length);
+		jsonStringBuilder.append(",");
+		jsonStringBuilder.append("\"synonyms\":");
+		int i = 0;

```

Comment:
@jdamerow  The variable i is used for checking if the synonym we are adding is the last in the synonym list. If that is the last synonym I wont be appending comma to the array in JSON.